runner.go

```diff
@@ -234,6 +243,12 @@ func filterExecuteTests(tests []*rainforest.RFTest, tags []string, forceExecute,
 	return result
 }
 
+func augmentJunitFileName(junitFile string, retryAttempt int) (string, error) {
+	pattern := regexp.MustCompile(`(\.[0-9])*\.xml`)
```

Comment:
This assumes we'll always retry in the single digits, so is broken for larger values.

I think it's broken for cases like `.0.1.2.3.4.5.xml` as well. Because `(foo)*` means match the thing inside the parenthesis zero or more times?
```suggestion
	pattern := regexp.MustCompile(`(\.[0-9]+)?\.xml$`)
```

TBH - I think it'd be cleaner to just append the retry attempt to the end of the filename and remove the constraint we've added about the file having to have an `.xml` suffix at 
https://github.com/rainforestapp/rainforest-cli/blob/2a2d3ad7576d097422c593d1bf5ed0723f16ca19/reporter.go#L93

This requirement is bad because:
1. maybe the user doesn't want to have a file ending in `.xml`
2. because the check is only made at the end of the run - so we could complete the run and then blow up and not save the junit output because the user didn't give a file name that ends in `.xml`!

Reply:
Good point, I'll get rid of the check and append `.1` to the end of the filename.

Reply:
> This requirement is bad because:
> 
>     1. maybe the user doesn't want to have a file ending in `.xml`

Is that really a thing? I feel like the opposite is more likely. Will Circle for example recognize a JUnit file if it doesn't have the `xml` extension? I can't tell from [the docs](https://circleci.com/docs/2.0/collect-test-data/).
 
>     2. because the check is only made at the end of the run - so we could complete the run and then blow up and not save the junit output because the user didn't give a file name that ends in `.xml`!

That's a very good point, but we could just move the check to before the run gets kicked off?

Reply:
> Will Circle for example recognize a JUnit file if it doesn't have the xml extension

I don't think that's a concern of the CLIs and we have specific tooling (the orb) to handle running the CLI in the context of CircleCI. The norms for unix CLIs don't expect filenames to mean much - the only one I can think of that I normally use is `unzip`, and I normally curse that for being un-ergonomic for exactly this reason.

I also know I would curse a utility if it forced me to have a filename of a particular format when I wanted it in another, because then I have to write stuff to handle renaming it. It also means we can't specify `/dev/fd/3` as an output to allow users to pipe the output somewhere else (as long as the FD functions we're using don't do anything funky, this should just work TM). It's just a needless restriction IMO.

> but we could just move the check to before the run gets kicked off

Sure - still don't think the CLI should enforce filenames.